src/DBAL/PingConnectionMiddleware.php

```diff
@@ -26,11 +31,30 @@ public function __construct(Connection $connection)
      */
     public function execute(object $command, callable $next)
     {
-        if (! $this->connection->ping()) {
+        if (! $this->ping($this->connection)) {
             $this->connection->close();
             $this->connection->connect();
         }
 
         return $next($command);
     }
+
+    private function ping(Connection $connection): bool
+    {
+        set_error_handler(static function (int $severity, string $message, string $file, int $line): bool {
```

Comment:
Silly question but wouldn't a method_exists check also work here and be less invasive/global state?

Reply:
Removed. It was added to handle error raised during connection, but now it can be handle by `Throwable`, so can be simplified. Proved by test!

Reply:
@rosstuck ping. I would like to finish this because this library it's last dbal v2 requirement.